CommonApi/src/main/java/mezz/jei/api/recipe/category/extensions/IGlobalRecipeCategoryExtension.java

```diff
@@ -0,0 +1,67 @@
+package mezz.jei.api.recipe.category.extensions;
+
+import mezz.jei.api.gui.drawable.IDrawable;
+import mezz.jei.api.gui.ingredient.IRecipeSlotsView;
+import mezz.jei.api.helpers.IGuiHelper;
+import mezz.jei.api.recipe.category.IRecipeCategory;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.network.chat.Component;
+
+import java.util.List;
+
+public interface IGlobalRecipeCategoryExtension<T> {
```

Comment:
This name is pretty long and generic, we should come up with a better name for this and its methods that help make it feel like a more concrete concept.
I'll try to help think of something.

Reply:
I agree. Sadly, `IRecipeCategoryExtension` is already taken and I couldn't come up with another name.
What about something like `ICategoryDecorator`?

Reply:
`IRecipeCategoryDecorator` sounds good to me